component/prometheus/remotewrite/types.go

```diff
@@ -89,7 +89,17 @@ func (r *EndpointOptions) UnmarshalRiver(f func(v interface{}) error) error {
 	*r = DefaultEndpointOptions
 
 	type arguments EndpointOptions
-	return f((*arguments)(r))
+	err := f((*arguments)(r))
+	if err != nil {
+		return err
+	}
+
+	// We must explicitly Validate because HTTPClientConfig is squashed and it won't run otherwise
+	if r.HTTPClientConfig != nil {
```

Comment:
This is noticeable different than the prometheus.scrape implementation but necessary because r.HTTPClientConfig is a pointer in prometheus.remote_write.